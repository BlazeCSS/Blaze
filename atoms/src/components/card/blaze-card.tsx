import { Component } from '@stencil/core';

@Component({
  tag: 'blaze-card'
})

export class Card {

  render() {
    return (
      <div class="c-text c-card">
        <slot name="image" />
        <header class="c-card__header">
          <slot name="heading" />
        </header>
        <div class="c-card__body">
          <slot name="body" />
        </div>
        <footer class="c-card__footer">
          <slot name="footer" />
        </footer>
      </div>
    );
  }
}
