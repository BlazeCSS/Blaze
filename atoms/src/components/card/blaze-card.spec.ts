import { flush, render } from '@stencil/core/testing';
import { Card } from './blaze-card';

describe('Card', () => {
  it('should build', () => {
    expect(new Card()).toBeTruthy();
  });

  let element;

  const snapIt = (name, html) => {
    it(name, async () => {
      element = await render({
        components: [Card],
        html
      });
      await flush(element);

      expect(element).toMatchSnapshot();
    });
  };

  snapIt(
    'renders a card',
    '<blaze-card></blaze-card>'
  );

  snapIt(
    'renders all the child elements correctly',
    `<blaze-card>
      <blaze-image></blaze-image>
      <blaze-card-header>
        <h2 class="c-heading u-xlarge">
          Heading
          <div class="c-heading__sub">Sub-heading</div>
        </h2>
      </blaze-card-header>
      <blaze-card-body>
        <p class="c-paragraph">
          Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.
        </p>
      </blaze-card-body>
      <blaze-card-footer>
        <div class="c-input-group">
          <button class="c-button c-button--block c-button--brand">Button</button>
          <button class="c-button c-button--block c-button--info">Button</button>
        </div>
      </blaze-card-footer>
    </blaze-card>`
  );
});
