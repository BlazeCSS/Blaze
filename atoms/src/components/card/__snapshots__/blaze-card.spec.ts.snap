// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders a card 1`] = `
<blaze-card
  class="hydrated"
>
  <div
    class="c-card"
  >
    <!---->
  </div>
</blaze-card>
`;

exports[`Card renders all the child elements correctly 1`] = `
<blaze-card
  class="hydrated"
>
  <div
    class="c-card"
  >
    <!---->
    
      
    <blaze-image />
    
      
    <blaze-card-header>
      
        
      <h2
        class="c-heading u-xlarge"
      >
        
          Heading
          
        <div
          class="c-heading__sub"
        >
          Sub-heading
        </div>
        
        
      </h2>
      
      
    </blaze-card-header>
    
      
    <blaze-card-body>
      
        
      <p
        class="c-paragraph"
      >
        
          Lorem ipsum dolor sit amet, feugiat corpora ex eam. Inciderint eloquentiam sea et.
        
      </p>
      
      
    </blaze-card-body>
    
      
    <blaze-card-footer>
      
        
      <div
        class="c-input-group"
      >
        
          
        <button
          class="c-button c-button--block c-button--brand"
        >
          Button
        </button>
        
          
        <button
          class="c-button c-button--block c-button--info"
        >
          Button
        </button>
        
        
      </div>
      
      
    </blaze-card-footer>
    
    
  </div>
</blaze-card>
`;
